### PR TITLE
Add `modernize-use-auto` clang-tidy check

### PR DESCRIPTION
**Pull request by ChrisThrasher**
## Description

All these changes should improve readability by avoiding writing really long type names or avoiding writing the type name twice on the same line.

### CODE REVIEW

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Review by eXpl0it3r — APPROVED** [review]
Looks good to me. Interested in hearing @Bromeon opinion, since he had some stronger opinion on `auto` usage in the past.

**Review by Bromeon — COMMENTED** [review]
Arguably it can make it a bit harder to recognize the type after this change, as one needs to look at the expression on the right-hand-side. It's one of these "nice to satisfy a tool" things; if truly necessary, I'm not sure. Admittedly it does remove repetition though, which is nice.

Imo `auto` is mainly problematic for things like `auto size = getSize(...)`, where it's no longer obvious whether we're talking about a vector or scalar, float or int, etc. Since the type here still appears, it shouldn't be a big issue.

TLDR: no strong opinion, if others find it more readable, let's go for it.

File: src/SFML/Window/OSX/HIDInputManager.mm [hunk 1]
```diff
@@ -58,7 +58,7 @@
     if (!typeRef || (CFGetTypeID(typeRef) != CFNumberGetTypeID()))
         return 0;
 
-    CFNumberRef locRef = static_cast<CFNumberRef>(typeRef);
+    const auto* locRef = static_cast<CFNumberRef>(typeRef);
```
[the remark this answers is not in the document]
Comment (ChrisThrasher) [reply src/SFML/Window/OSX/HIDInputManager.mm:61]: > Maybe pedantic, but is auto* not exposing an implementation detail of an opaque type?

Perhaps yes.

> Meaning, we could simply use auto as well.

That does work but the problem is there's another clang-tidy check that ensures that you use `auto*` with pointers. The compiler can properly deduce that it's a pointer without the `*` but for readability's sake clang-tidy has us add that asterisk.

[the remark this answers is not in the document]
Comment (Bromeon) [reply src/SFML/Window/OSX/HIDInputManager.mm:61]: That's kind of the problem with having many of these opinionated checks 🙈
Anyway, minor issue, thanks for explanation!
